src/main/java/io/stargate/sgv2/jsonapi/service/shredding/model/JsonExtensionType.java

```diff
@@ -0,0 +1,59 @@
+package io.stargate.sgv2.jsonapi.service.shredding.model;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Types that "extend" JSON scalar type: by wrapping scalar value (usually String) in a single-entry
+ * JSON Object value where key indicates the type. These are generally used to retain client-side
+ * type information in cases where processing on server-side is still based on enclosed String
+ * value; one exception being "EJSON Date" which is used to specify Date values for indexing and
+ * sort purposes.
+ */
+public enum JsonExtensionType {
+  EJSON_DATE("$date"),
```

Comment:
Because that is the only type covered by EJSON spec (although we are not really implementing that spec in any case). Others are simple wrapped types we are selecting.